File: app/src/main/java/com/mywidget/Interface/LmemoApi.java
package com.mywidget.Interface;

import com.google.gson.JsonObject;
import io.reactivex.Observable;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface LmemoApi {

    String URL = "https://datewidget-ab4ba.firebaseio.com/";

    @GET("Lmemo/{name}.json")
    Call<JsonObject> lmemoData(@Path("name") String name);

    @GET("Lmemo/{name}.json")
    Observable<JsonObject> lmemoData2(@Path("name") String name);
}

File: app/src/main/java/com/mywidget/data/Interface/LmemoApi.kt
package com.mywidget.data.Interface

import com.google.gson.JsonObject
import com.mywidget.data.model.UserData
import io.reactivex.Observable
import org.json.JSONObject
import retrofit2.Response
import retrofit2.http.*
import java.util.*

interface LmemoApi {

    @GET("Lmemo/{name}.json")
    fun lmemoData(@Path("name") name: String): Observable<JsonObject>

    @GET("User/{id}.json")
    fun UserData(@Path("id") id: String?): Observable<UserData>

    @POST
    fun fcmTest(@Url url: String, @HeaderMap map: HashMap<String, String>, @Body body: JSONObject): Observable<Response<*>>
}
